front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/ChartHelpers.ts

```diff
@@ -320,12 +320,10 @@ export const interpolateOnTime = (
     let bisection;
     // If not array of array
     if (listValue === 'speed' || listValue === 'speeds') {
-      if (
-        dataSimulation?.[listValue] &&
-        dataSimulation?.[listValue][0] &&
-        timePositionLocal >= dataSimulation?.[listValue][0][keyValues[0]]
-      ) {
-        const index = bisect(dataSimulation[listValue], timePositionLocal, 1);
+      const comparator = dataSimulation?.[listValue] || dataSimulation?.[listValue][0]
```

Comment:
why did the `&&` become a `||` 

if `dataSimulation?.[listValue]` is truthy you never enter the else if line 331

Reply:
DataSimulation can be wrongly typed, which is not the topic of this PR. It can ever be an array of value if listValue is (speed | speeds) . But we can create an issue on that matter

Reply:
Add missing Types for SimulationResults/ChartHelpers/ChartHelpers.ts

Reply:
Ok ! we’ll type this file better. 
https://github.com/orgs/DGEXSolutions/projects/6/views/25?pane=issue&itemId=19286176